addon/utils.js

```diff
@@ -97,6 +100,79 @@ export function generateFacetView (facets) {
   }
 }
 
+/**
+ * Get merged definition for current cell
+ * @param {BunsenCell} cellConfig - current cell
+ * @param {Object<String, BunsenCell>} cellDefinitions - list of cell definitions
+ * @returns {BunsenCell} current merged cell definition
+ */
+export function getMergedConfig (cellConfig, cellDefinitions) {
+  if (!cellConfig.extends) {
+    return _.cloneDeep(cellConfig)
+  }
+
+  const superCell = getMergedConfig(cellDefinitions[cellConfig.extends], cellDefinitions)
+  const mergedConfig = assign(superCell, cellConfig)
+
+  delete mergedConfig.extends
+
+  return mergedConfig
+}
+
+export function getRendererComponentName (rendererName) {
+  return builtInRenderers[rendererName] || rendererName
+}
+
+/**
+ * Determine if last segment of a bunsen model path is required
+ * @param {String} path - path to property in bunsen model
+ * @param {BunsemModel} bunsenModel - bunsen model
+ * @returns {Boolean} whether or not last path is required
+ */
+export function isLastSegmentRequired (path, bunsenModel) {
+  const segments = path.split('.')
+  const lastSegment = segments.pop()
+
+  // Make sure we get the correct bunsen model for nested properties
+  if (segments.length !== 0) {
+    bunsenModel = get(bunsenModel, getModelPath(segments.join('.'))) || bunsenModel
+  }
```

Comment:
In general, I dislike seeing comments like line 136 and 141 in code because (1) if the code isn't readable enough to require a comment, then that's a code smell, and (2) comments like this are not maintained to the same level of attention as the code itself is maintained. In other words, you could easily see some developer coming in later and refactoring this code but leaving that comment (because most devs just ignore comments when working on code), and you could easily see that comment in six months being totally irrelevant and wrong. Instead of putting a comment like that in the code, I find it generally better to factor out a function with the same name as the comment.

For example, what about:
```
function getCorrectBunsenModelForNestedProperties(bunsenModel, segments) {
  if (segments.length !== 0) {
    bunsenModel = get(bunsenModel, getModelPath(segments.join('.'))) || bunsenModel
  }
  return bunsenModel
}
....

export function isLastSegmentRequired (path, bunsenModel) {
  const segments = path.split('.')
  const lastSegment = segments.pop()
  bunsenModel = getCorrectBunsenModelForNestedProperties(bunsenModel, segments)
  ...
}
```